### PR TITLE
fix msal token cache issue

### PR DESCRIPTION
**Pull request by backwind1233**
- msal lib will cache token and the cache will not be removed automatically by default
- to fix this partially, do this refactor

### CODE REVIEW

**Review by saragluna — APPROVED** [review]
LGTM.

**Review by saragluna — APPROVED** [review]
LGTM.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java [hunk 1]
```diff
@@ -41,16 +38,11 @@
...
-        this.aadWebApiClientRegistrationRepository = aadWebApiClientRegistrationRepository;
-        Iterator<ClientRegistration> iterator = aadWebApiClientRegistrationRepository.iterator();
-        while (iterator.hasNext()) {
-            ClientRegistration next = iterator.next();
-            this.confidentialClientApplicationMap.put(next.getRegistrationId(), createApp(next));
-        }
+      AADWebApiClientRegistrationRepository azureClientRegistrationRepository) {
+      
+        this.azureClientRegistrationRepository = azureClientRegistrationRepository;
     }
```
Comment (wujack778) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java:46]: commit something

**Review by wujack778 — COMMENTED** [review]
comment

**Comment by saragluna** [comment]
Closing this PR since we've merged this in #18257.

**Pull request closed without merging**